ansible_base/authentication/serializers/authenticator_map.py

```diff
@@ -17,17 +18,42 @@ def validate(self, data) -> dict:
         map_type = data.get('map_type', None)
         team = data.get('team', None)
         org = data.get('organization', None)
+        role = data.get('role', None)
+
         if map_type == 'team' and (not team or team == ''):
             errors["team"] = "You must specify a team with the selected map type"
-        if map_type == 'team' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
-        if map_type == 'organization' and (not org or org == ''):
+        if map_type in ['team', 'organization'] and (not org or org == ''):
             errors["organization"] = "You must specify an organization with the selected map type"
+        if map_type in ['team', 'organization', 'role'] and role is None:
+            errors["role"] = "You must specify a role with the selected map type"
+        if map_type in ['allow', 'is_superuser'] and role is not None:
+            errors["role"] = "You cannot specify role with the selected map type"
+
+        if role is not None:
+            errors.update(self.validate_role_data(role, org, team))
 
         if errors:
             raise ValidationError(errors)
         return data
 
+    def validate_role_data(self, role, org, team):
+        errors = {}
+
+        if not role.content_type:
+            return errors
+
+        model_class = role.content_type.model_class()
+        is_org_role = model_class == get_organization_model()
+        is_team_role = model_class == get_team_model()
```

Comment:
How does `==` compare to `isinstance`?

Reply:
this works well, because the class and `get_*` are derived from settings. 
moreover they are classes, not instances, so `isinstance` won't work. But I can use `issubclass` for more clean solution